jerry-core/vm/vm.c

```diff
@@ -620,11 +619,26 @@ vm_init_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
                                                      literal_start_p[value_index]);
           }
 
-          /* TODO: check the return value */
-          ecma_op_put_value_lex_env_base (ref_base_lex_env_p,
-                                          name_p,
-                                          is_strict,
-                                          lit_value);
+          if ((ecma_get_lex_env_type (frame_ctx_p->lex_env_p) == ECMA_LEXICAL_ENVIRONMENT_DECLARATIVE)
```

Comment:
That check is added because `value_index == const_literal_end` check wasn't enough for global environment code. The `value_index` of first function declaration in global environment was same as `const_literal_end`, so I wanted to exclude the global environment.
Anyway since it seems to be hard to understand the code, I am trying to modify this check code.